src/App.jsx

```diff
@@ -14,33 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, { sortField, query, reversed }) {
+  let preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (query) {
+    preparedGoods = preparedGoods.filter(good => good.includes(query));
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField) {
+    preparedGoods.sort((g1, g2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return g1.localeCompare(g2);
+
+        case SORT_FIELD_LENGTH:
+          return g1.length - g2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.toReversed();
```

Comment:
The method `toReversed()` is not a valid JavaScript array method. You should use `reverse()` instead to reverse the array in place. However, note that `reverse()` mutates the original array, so you might want to use `slice().reverse()` to avoid mutating the original array.